scripts/west_commands/run_common.py

```diff
@@ -134,9 +136,21 @@ def add_parser_common(command, parser_adder=None, parser=None):
     group.add_argument('--board-dir', metavar='DIR', help='board directory')
     # FIXME: we should just have a single --file argument. The variation
     # between runners is confusing people.
-    group.add_argument('--elf-file', metavar='FILE', help='path to zephyr.elf')
-    group.add_argument('--hex-file', metavar='FILE', help='path to zephyr.hex')
-    group.add_argument('--bin-file', metavar='FILE', help='path to zephyr.bin')
+    group.add_argument('--elf-file',
+                       metavar='FILE',
+                       action=partial(depr_action,
+                                      replacement='-f/--file'),
+                        help='path to zephyr.elf')
+    group.add_argument('--hex-file',
+                       metavar='FILE',
+                       action=partial(depr_action,
+                                      replacement='-f/--file'),
```

Comment:
Doing this unconditionally means that `west flash -r nrfjprog --hex-file some-other.hex` now prints a warning, despite that runner not supporting the file capability. This is not desirable.

Reply:
that's a good point, but what is the best way to do it?
- move the `--xxx_file` arguments into each runner that doesn't support `--file` yet
- create the deprecation message in every runner that supports `--file''
- add a way to get runner caps in `add_parser_common()` to make the `depr_action` conditional or remove the `--xxx_file` arguments

Reply:
Looking lower down in the file:

``py
    add_parser_common(command, parser=parser)
    runner_cls.add_parser(parser)
```

It seems like you could move the decision making for these arguments into `ZephyrBinaryRunner.add_parser()`, which is a place where the `RunnerCaps` object is available for use in decision making.

That would let you only add the deprecation message for runners with the `file` capability. I would modify the deprecation message to something more like `f"option {deprecated_option} is deprecated for your runner {cls.name()}; use --file"`. Then we can begin a transition campaign over to use `--file` exclusively in all the runners at some point.